subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/MissingTaskDependenciesFixture.groovy

```diff
@@ -20,9 +20,9 @@ import groovy.transform.SelfType
 
 @SelfType(AbstractIntegrationSpec)
 trait MissingTaskDependenciesFixture {
-    void expectMissingDependencyDeprecation(String producer, String consumer, File producedConsumedLocation) {
+    void expectMissingDependencyDeprecation(String producer, String consumer, File producedConsumedLocation, String taskName = 'DefaultTask') {
         executer.expectDocumentedDeprecationWarning(
-            "Gradle detected a problem with the following location: '${producedConsumedLocation.absolutePath}'. " +
+            "Type '$taskName': Gradle detected a problem with the following location: '${producedConsumedLocation.absolutePath}'. " +
```

Comment:
I don't think the type name adds any useful information to this warning. Can we only show the type name for the problems with task class validation, and not for this runtime validation?

Reply:
It depends whether the problem is in the task implementation itself or not, right? If it's a problem in how it's configured, then you don't really need the type, but if it's the task misconfigured itself (say with a wrong convention then it's useful to have the task type name).

Reply:
A task never should reference another task directly, the wiring should be done by the plugin creating the task. So the task class doesn't have much influence on it, does it? Moreover, the warning involves two tasks, and we don't even know if the type is the type of the consumer or the type of the producer. We either need to clarify that, or leave out the type. I would leave out the type.

Reply:
I currently have no way to leave out the type for this specific case. I can't make a difference, which is why the type context was missing from other deprecation warnings, making them useless.

Reply:
I think we should allow leaving out the type. I don't think we need to do this in the current PR.

I find the type particularly confusing for this deprecation warning, since it adds a non-symmetry: When the producer runs first, then the reported type will be that of the consumer, and when the consumer runs first, then the reported type will be that of the producer. So we would have two different messages for the same problem.

Reply:
Ok I created https://github.com/gradle/gradle/issues/16255